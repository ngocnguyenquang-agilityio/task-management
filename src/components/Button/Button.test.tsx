import { render, screen, fireEvent } from '@testing-library/react';

// Components
import { Button } from './';

describe('Button component', () => {
  test('renders the Button with children', () => {
    render(<Button>Click me</Button>);
    expect(screen.getByText('Click me')).toMatchSnapshot();
  });

  test('renders startIcon and endIcon', () => {
    render(
      <Button startIcon={<span>Start</span>} endIcon={<span>End</span>}>
        Click me
      </Button>,
    );
    expect(screen.getByText('Start')).toBeInTheDocument();
    expect(screen.getByText('End')).toBeInTheDocument();
  });

  test('applies the correct variant classes', () => {
    render(<Button variant="outline">Click me</Button>);
    expect(screen.getByText('Click me')).toHaveClass(
      'border border-gray-200 hover:bg-gray-100 bg-transparent text-black',
    );
  });

  test('applies customClass', () => {
    render(<Button customClass="custom-class">Click me</Button>);
    expect(screen.getByText('Click me')).toHaveClass('custom-class');
  });

  test('handles click events', () => {
    const handleClick = jest.fn();
    render(<Button onClick={handleClick}>Click me</Button>);
    fireEvent.click(screen.getByText('Click me'));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  test('is disabled when the disabled prop is set', () => {
    render(<Button disabled>Click me</Button>);
    expect(screen.getByText('Click me')).toBeDisabled();
  });

  test('renders button with type submit', () => {
    render(<Button type="submit">Submit</Button>);
    expect(screen.getByText('Submit')).toHaveAttribute('type', 'submit');
  });
});
