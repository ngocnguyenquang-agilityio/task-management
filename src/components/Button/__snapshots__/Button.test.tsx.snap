// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders the Button with children 1`] = `
<button
  class="flex items-center rounded-lg p-2 disabled:cursor-not-allowed bg-neutral-800 text-white hover:bg-neutral-700 dark:bg-black dark:hover:opacity-70 disabled:bg-neutral-800/50"
  type="button"
>
  Click me
</button>
`;
